recipe/meta.yaml

```
@@ -1,124 +1,166 @@
+{% set number = "4" %}
+{% set cuda_enabled = cuda_compiler_version is not undefined and cuda_compiler_version == '9.2' %}
+{% set build_ext_version = "1.0.0" %}
+{% set build_ext = "cuda" if cuda_enabled else "cpu" %}
 
 package:
-  name: arrow-cpp
-  version: {{ version }}
+  name: arrow-cpp-ext
 
 source:
   fn: {{ filename }}
   url: https://dist.apache.org/repos/dist/release/arrow/arrow-{{ version }}/{{ filename }}
   sha256: {{ checksum }}
 
 build:
-  number: 3
-  skip: true  # [win and vc<14]
+  skip: true  # [(win and vc<14) or (linux and cuda_compiler_version is not undefined and cuda_compiler_version != '9.2' and cuda_compiler_version != 'None')]
+  number: {{ number }}
   run_exports:
-    - {{ pin_subpackage("arrow-cpp", max_pin="x.x.x") }}
+    - {{ pin_subpackage("arrow-cpp", max_pin="x.x.x") }}  # is this correct and the correct place for it?
 
-requirements:
-  build:
-    - cmake
-    - autoconf  # [unix]
-    - ninja
-    - make  # [unix]
-    - {{ compiler('c') }}
-    - {{ compiler('cxx') }}
-  host:
-    - aws-sdk-cpp
-    - boost-cpp
-    - brotli
-    - bzip2
-    - c-ares
-    - gflags
-    - glog
-    - grpc-cpp
-    - libprotobuf
-    - clangdev 7.*  # [not (aarch64 or ppc64le)]
-    - llvmdev 7.*   # [not (aarch64 or ppc64le)]
-    - lz4-c
-    - numpy 1.16.*
-    - orc  # [unix]
-    - python
-    - rapidjson
-    - re2
-    - snappy
-    - thrift-cpp >=0.11
-    - zlib
-    - zstd
+outputs:
+  - name: arrow-cpp-proc
+    version: {{ build_ext_version }}
+    build:
+      number: 0
+      string: "{{ build_ext }}"
+    test:
+      commands:
+        - exit 0
+    about:
+      home: http://github.com/apache/arrow
+      license: Apache-2.0
+      license_file:
+        - LICENSE.txt
+      summary: 'A meta-package to select Arrow build variant'
 
-  run:
-    - {{ pin_compatible('numpy', lower_bound='1.16') }}
-    - aws-sdk-cpp
-    - boost-cpp
-    - brotli
-    - c-ares
-    - gflags
-    - glog
-    - grpc-cpp
-    - lz4-c
-    - python
-    - re2
-    - snappy
-    - zlib
-    - zstd
+  - name: arrow-cpp
+    script: build-arrow.sh  # [not win]
+    script: bld.bat         # [win]
+    version: {{ version }}
+    build:
+      number: {{ number }}
+      string: "py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ number }}_{{ build_ext }}"
+      skip: true  # [win and vc<14]
+      skip: true  # [linux and cuda_compiler_version is not undefined and cuda_compiler_version != '9.2' and cuda_compiler_version != 'None']
+      ignore_run_exports:
+        - cudatoolkit
+      track_features:
+        - arrow-cuda         # [linux and cuda_compiler_version is not undefined and cuda_compiler_version == '9.2']
+    requirements:
+      build:
+        - cmake
+        - autoconf  # [unix]
+        - ninja
+        - make      # [unix]
+        - {{ compiler('c') }}
+        - {{ compiler('cxx') }}
+        - {{ compiler("cuda") }}  # [linux and cuda_compiler_version is not undefined and cuda_compiler_version == '9.2']
```

Since this already uses Jinja, we might consider condensing this all into Jinja.

```suggestion
        {{ ("-" ~ compiler("cuda")) if cuda_enabled else "" }}
```

---

Fixed.

---

This is a bad idea. conda-build's variant detection is text based and this interferes with used variable detection as seen in https://github.com/conda-forge/arrow-cpp-feedstock/pull/125/commits/bf07c1a3d3ff9c6dc2e0fa743ea15233e729cb1e where `cuda_compiler` variable is removed.

---

Hmm...that's unfortunate. Thanks for catching though Isuru. Was trying to find a way to simplify the selection logic, but agree conda-build makes that a bit tricky today.